platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/AbstractCollectionProperty.java

```diff
@@ -486,4 +497,61 @@ public ValueProducer getProducer() {
             return left.getProducer().plus(right.getProducer());
         }
     }
+
+    protected CollectionSupplierGuard<T, C> guardSupplier(CollectionSupplier<T, C> supplier) {
```

Comment:
💅 It could be just `guard` in this context.

Reply:
Done